app/Http/Controllers/AtcBookingController.php

```diff
@@ -0,0 +1,106 @@
+namespace App\Http\Controllers;
+
+use App\AtcBooking;
+use Auth;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+class AtcBookingController extends Controller {
+    public function viewBookings() {
+        $today = Carbon::today();
+        $bookings = AtcBooking::whereDate('start', '>=', $today)
+            ->where(function ($query) use ($today) {
+                $query->whereDate('end', '<=', $today->addDays(14))
+                      ->orWhere('cid', Auth::id());
+            })
+            ->orderBy('start', 'ASC')
+            ->get();
+
+        $bookings = groupAtcBookingsByDate($bookings);
+
+        $types = [
+            AtcBooking::TYPES['BOOKING'] => "Booking"
+        ];
+
+        if (Auth::user()->isAbleTo('train') || Auth::user()->isAbleTo('snrStaff')) {
+            $types[AtcBooking::TYPES['MONITORING']] = "Monitoring";
+        }
+
+        if (Auth::user()->isAbleTo('events') || Auth::user()->isAbleTo('snrStaff')) {
+            $types[AtcBooking::TYPES['EVENT']] = "Event";
+        }
+
+        if (Auth::user()->hasRole('ins') || Auth::user()->isAbleTo('snrStaff')) {
+            $types[AtcBooking::TYPES['EXAM']] = "Exam";
+        }
+
+        return view('dashboard.controllers.bookings.view_bookings')->with('bookings', $bookings)->with('types', $types);
+    }
+
+    public function createBooking(Request $request) {
+        $validator = $request->validate([
+            'facility' => 'required',
+            'position' => 'required',
+            'type' => 'required',
+            'start' => 'required|date_format:m/d/Y H:i|after:now',
+            'end' => 'required|date_format:m/d/Y H:i|after:start',
+        ]);
+
+        $callsign = $request->facility . "_" . $request->position;
+        $start = Carbon::createFromFormat('m/d/Y H:i', $request->start)->toDateTimeString();
+        $end = Carbon::createFromFormat('m/d/Y H:i', $request->end)->toDateTimeString();
+
+        $existing = AtcBooking::where('callsign', $callsign)
+                              ->where(function ($query) use ($start, $end) {
+                                  $query->where(function ($query) use ($start) {
+                                      $query->where('start', '>=', $start)
+                                              ->where('end', '<', $start);
+                                  })
+                                     ->orWhere(function ($query) use ($end) {
+                                         $query->where('start', '<', $end)
+                                                 ->where('end', '>=', $end);
+                                     });
+                              })->count() > 0;
+
+        if ($existing) {
+            return redirect()->back()->with('error', 'A booking already exists for ' . $callsign . ' at this time')->withInput();
+        }
+
+        $type = $request->type;
+        if ($type == AtcBooking::TYPES["EVENT"] && ! (Auth::user()->hasRole('ec') || Auth::user()->isAbleTo('snrStaff'))) {
```

Comment:
Please change this from EC to events team... I think the permission is `isAbleTo('events')`

Reply:
Resolved and rebased into 32b9f26